app/controllers/sessions_controller.rb

```diff
@@ -1,5 +1,13 @@
 class SessionsController < Devise::SessionsController
 
+  def initialize
+    @announcements = Announcement
+      .where(on_main_page: true, lecture: nil)
+      .pluck(:details)
+      .join
+    super
+  end
+
```

Comment:
I added the announcements to the landing page view (right now called `devise.html.erb`). The problem is that when I set the `@announcements` variable here in the `SessinosController`, announcements will only show up for the `/users/sign_in` page, not even the landing page. So, I could just go ahead and add this to the devise registration controller and confirmation controller, but this seems repetitive (not DRY), and this won't solve the problem that it doesn't work for the landing page that gets mapped to the `devise/sessions#new` controller in the `unauthenticated` block in `routes.rb`.

What we would need is a controller that handles the landing page on its own. The corresponding view should have the devise layout as nested layout. However, I don't know how to do that, especially with the controller, and I can't figure this out after endless searching. These might be some links to consider:

- [devise `authenticate` method](https://github.com/heartcombo/devise/blob/3926e6d9eb139cc839faec8ea6c8f8cefa2d95f6/lib/devise/rails/routes.rb#L296-L334)
- [devise configuring controllers](https://github.com/heartcombo/devise#configuring-controllers)
- [Devise: Sign Up Page as Welcome/Landing Page then to User Profile](https://stackoverflow.com/questions/5825135/devise-sign-up-page-as-welcome-landing-page-then-to-user-profile)]
- [How to render a view inside of another view (rails 4)](https://stackoverflow.com/questions/19844652/how-to-render-a-view-inside-of-another-view-rails-4)

Reply:
Thanks, that works ;)